### ISSUE TITLE
MKL instead of OpenBLAS on OSX (intel)

### ISSUE THREAD

**Issue by DoctorDro** [JuliaSmoothOptimizers/HSL.jl#86]
Dear all,
I came across a weird problem trying to replace OpenBLAS with MKL. Although I have installed the MKL package (] add MKL ; test MKL), in deps/build.jl I added as first statement ( using MKL ). But when the code runs it complains that:

>(@v1.7) pkg> build --verbose HSL
    Building HSL → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/bfe35eb36d387c1d0913aa2359210dac32764e19/build.log`
ERROR: LoadError: ArgumentError: Package MKL not found in current path:
- Run `import Pkg; Pkg.add("MKL")` to install the MKL package.

>Stacktrace:
 [1] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:967
 [2] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [3] top-level scope
   @ none:5

Any ideas?

**Comment by amontoison** [comment]
Hi @DoctorDro, If you want to replace OpenBLAS with MKL you need the artifact `MKL_jll` and compile HSL with it.
We first need to add it in the Project.toml: https://github.com/JuliaSmoothOptimizers/HSL.jl/blob/main/Project.toml#L18
and compile with the library of MKL here: https://github.com/JuliaSmoothOptimizers/HSL.jl/blob/main/deps/build.jl#L86

You can't easily switch between OpenBLAS and MKL.

**Comment by amontoison** [comment]
@DoctorDro 
You should be able to use MKL as BLAS backend after #87.

**Comment by dpo** [comment]
Please open a new issue if problems persist.
